src/common/bots/Bot.cpp

```diff
@@ -0,0 +1,110 @@
+        }
+    );
+
+    bool go_left = closestAllowedDirection.x == -1;
+    bool go_right = closestAllowedDirection.x == 1;
+    bool go_up = closestAllowedDirection.y == -1;
+    bool go_down = closestAllowedDirection.y == 1;
+    InputDef::UnreliableInput unreliable_input = {
+        .toggle_up = go_up,
+        .toggle_down = go_down,
+        .toggle_right = go_right,
+        .toggle_left = go_left,
+        .toggle_stop = false,
+    };
+    return std::pair<InputDef::ReliableInput, InputDef::UnreliableInput>(reliable_input,
+                                                                         unreliable_input);
+}
+
+/* This strategy will move towards the closest mine which has a resource the bot needs to win.
+ * Once it has enough of a resource, it will move to the next closest mine with a resource it needs. */
+std::pair<InputDef::ReliableInput, InputDef::UnreliableInput>
+Bot::getNearestGreedyMove(std::vector<std::shared_ptr<Mine>> mines,
+                std::array<uint32_t, RESOURCE_TYPE_COUNT> current_resources,
+                sf::Vector2f bot_center,
+                std::unordered_map<uint32_t, uint32_t> mine_id_to_resource_count
+                ) {
+
+
+    // how expensive is this? We really only need to do it when we change direction, since
+    // if we know we've been moving towards a mine we can assume we're continuing to move towards it.
+    // in which case we can't possibly have gotten closer to a different mine.
+    std::sort(mines.begin(), mines.end(), 
+            [bot_center] (std::shared_ptr<Mine> mine_a, std::shared_ptr<Mine> mine_b) {
+                auto dist_a = VectorUtil::distance(bot_center, mine_a->getCenter());
+                auto dist_b = VectorUtil::distance(bot_center, mine_b->getCenter());
+                return dist_a < dist_b;
+            }
+    );
+    for ( auto mine : mines ) {
+        auto curr_mine_id = mine->getId();
+        // this feels gross, but only because we haven't actually implemented unique  win conditions
+        if (mine_id_to_resource_count[curr_mine_id] != WIN_CONDITION[curr_mine_id]) {
+            // move towards this mine since it's closest and has a resource we need
+            auto direction_to_mine = VectorUtil::getVector(bot_center, mine->getCenter());
+            return vectorToNearestMoveTransform(direction_to_mine);
+        }
+    }
+}
+
+std::pair<InputDef::ReliableInput, InputDef::UnreliableInput>
+Bot::getRandomMove() {
+    InputDef::ReliableInput reliable_input;
+    reliable_input.setAll(false);
+    // run bot getMove(gameState) function
+    // return move in m_reliableInput and m_unreliableInput
+    srand(time(0));
+    bool up = bool(rand() % 2);
+    bool right = bool(rand() % 2);
+    InputDef::UnreliableInput unreliable_input = {
+        .toggle_up = up,
+        .toggle_down = !up,
+        .toggle_right = right,
+        .toggle_left = !right,
+        .toggle_stop = false,
+    };
```

Comment:
If I'm not mistaken, this bot can only go in diagonal directions...
Logic is off, you should be able to have `toggle_up` be true and everything else be false for example.

Reply:
Stopping is also a random behavior 
If players can stand still, mr rando 🤖  should be able to stand still too ;)

Reply:
good point on mr rando, he definitely didnt get the love he deserves